adapters/Unity/UnityAdapter/GADMAdapterUnityRewardedAd.m

```diff
@@ -117,70 +115,69 @@ - (NSString *)getPlacementID {
 #pragma mark - Unity Delegate Methods
 
 - (void)unityAdsDidError:(UnityAdsError)error withMessage:(nonnull NSString *)message {
-  if (!_isLoading) {
-    if (error == kUnityAdsErrorShowError) {
-      NSError *presentError = GADUnityErrorWithDescription(message);
-      [_adEventDelegate didFailToPresentWithError:presentError];
-    }
-    return;
-  }
-
-  if (_adLoadCompletionHandler) {
-    NSError *errorWithDescription = GADUnityErrorWithDescription(message);
-    _adLoadCompletionHandler(nil, errorWithDescription);
-    _adLoadCompletionHandler = nil;
-  }
-  _isLoading = NO;
 }
 
-- (void)unityAdsDidFinish:(nonnull NSString *)placementId
+- (void)unityAdsDidFinish:(nonnull NSString *)placementID
           withFinishState:(UnityAdsFinishState)state {
   id<GADMediationRewardedAdEventDelegate> strongDelegate = _adEventDelegate;
-  if (state == kUnityAdsFinishStateCompleted) {
-    [strongDelegate didEndVideo];
-    // Unity Ads doesn't provide a way to set the reward on their front-end. Default to a reward
-    // amount of 1. Publishers using this adapter should override the reward on the AdMob
-    // front-end.
-    GADAdReward *reward = [[GADAdReward alloc] initWithRewardType:@""
-                                                     rewardAmount:[NSDecimalNumber one]];
-    [strongDelegate didRewardUserWithReward:reward];
+  if (strongDelegate && [placementID isEqualToString:_placementID]) {
+    if (state == kUnityAdsFinishStateCompleted) {
+      [strongDelegate didEndVideo];
+      // Unity Ads doesn't provide a way to set the reward on their front-end. Default to a reward
+      // amount of 1. Publishers using this adapter should override the reward on the AdMob
+      // front-end.
+      GADAdReward *reward = [[GADAdReward alloc] initWithRewardType:@""
+                                                         rewardAmount:[NSDecimalNumber one]];
+      [strongDelegate didRewardUserWithReward:reward];
+    } else if (state == kUnityAdsFinishStateError) {
+      [strongDelegate willPresentFullScreenView];
+      NSError *presentError = GADUnityErrorWithDescription(@"Finish State Error");
```

Comment:
In the error state, is the "start" callback not called? Or could this be any error during presentation?

Wer want to make sure we only call willPresentFullScreenView once.